src/array_api_extra/_lib/_typing.py

```diff
@@ -19,4 +21,4 @@ def no_op_decorator(f):  # pyright: ignore[reportUnreachable]
 
 __all__ = ["ModuleType", "override"]
 if typing.TYPE_CHECKING:
-    __all__ += ["Array", "Device"]
+    __all__ += ["Array", "Device", "Index", "Untyped"]
```

Comment:
I see not advantage of putting this in a `TYPE_CHECKING` block. And I can imagine that mypy wouldn't know how to deal with this, but I might be wrong here.